src/events/dns.js

```diff
@@ -0,0 +1,358 @@
+        }
+        if (!rrtype) {
+            rrtype = rrtypesMethods.resolve4;
+        }
+        if (!arg3) {
+            callback = arg2;
+        }
+    }
+    return { hostname, rrtype, callback };
+};
+
+/**
+ * Gettign lookup arguments with options.
+ * @param {*} arg1 function first argument.
+ * @param {*} arg2 function second argument
+ * @param {*} arg3 function third argument
+ * @returns {Object} Object of Hostname, options (if doesnt exist- undifiend) and callback.
+ */
+const getLookupArguments = (arg1, arg2, arg3) => {
+    const hostname = arg1;
+    let options = arg2;
+    let callback = arg3;
+    if (!arg3) {
+        options = undefined;
+        callback = arg2;
+    }
+    return { hostname, options, callback };
+};
+
+/**
+ * Gettinng callback object of dns resolve.
+ * @param {*} arg callback argument
+ * @param {string} rrtype resouece record type
+ * @returns {Object} Returns the callback object with the appropriate key compared to rrtype,
+ * (if doesnt exist - records).
+ */
+const getCallbackResolveArgument = (arg, rrtype) => {
+    if (!arg) return undefined;
+    let callbackArg;
+    if (!rrtype || rrtype === rrtypesMethods.resolveTxt) {
+        callbackArg = { records: arg };
+    } else if (rrtype === rrtypesMethods.resolveAny) {
+        callbackArg = { ret: arg };
+    } else if (rrtype === rrtypesMethods.resolveSoa) {
+        callbackArg = { address: arg };
+    } else {
+        callbackArg = { addresses: arg };
+    }
+    return callbackArg;
+};
+
+/**
+ * Calling to the dns function without callback, And record the error if thrown.
+ * @param {Function} original dns function.
+ * @param {number} startTime Event start time.
+ * @param {serverlessEvent.Event} dnsEvent Dns event.
+ * @param {Array} args Array of function arguments.
+ * @returns {Object} original function response.
+ */
+function handleFunctionWithoutCallback(original, startTime, dnsEvent, args) {
+    let clientRequest;
+    try {
+        clientRequest = original.apply(this, args);
+    } catch (err) {
+        dnsEvent.setDuration(utils.createDurationTimestamp(startTime));
+        eventInterface.setException(dnsEvent, err);
+        tracer.addEvent(dnsEvent);
+        throw err;
+    }
+    return clientRequest;
+}
+
+/**
+ *  Wrap dns resolve requset
+ * @param {Function} original The dns function.
+ * @returns {Function} The wrapped function
+ */
+function wrapDnsResolveFunction(original) {
+    return function internalWrapDnsResolveFunction(arg1, arg2, arg3) {
+        let patchedCallback;
+        let clientRequest;
+        let options;
+        const { hostname, rrtype, callback } = getRrtypeArguments(arg1, arg2, arg3, original.name);
+        if (typeof arg2 === 'object') {
+            options = arg2;
+        }
+        const { dnsEvent, startTime } = initialDnsEvent(original.name);
+        if (!callback) {
+            clientRequest = handleFunctionWithoutCallback(
```

Comment:
I would just make this `return handleFunctionWithoutCallback` so it's obvious that execution doesn't continue after this point

Reply:
this way you don't need the `else` block

Reply:
nice